### PR TITLE
Add a docker compose file

### PR DESCRIPTION
**Pull request by dngray**
It's quite the norm to include a docker-compose file, generally in the README or the root for people to copy and modify. For example as https://github.com/wfg/docker-openvpn-client has done so.

If there are [Environmental variables](https://github.com/wfg/docker-openvpn-client#environment-variables), they should also be documented - in this case there isn't.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  protonmail-bridge:
+    image: shenxn/protonmail-bridge
+    # build:
+    #  context: ./build
+    #  dockerfile: Dockerfile
+    container_name: pm_bridge
+    ports:
+      - 1025:25/tcp
+      - 1143:143/tcp
+    volumes:
```
Comment (eyduh) [inline docker-compose.yml:11]: this needs a volume section on the bottom like

```
volumes:
  protonmail:
    name: protonmail
```

**Review by eyduh — CHANGES_REQUESTED** [review]
It is good practise to have the docker-compose version stated at the top of the file, for compatibility with GUIs like portainer one might want to stick to 
`version: '2.1'`
 
Also, should add a volume declaration
```
volumes:
  protonmail:
    name: protonmail
```

I also added an external network to my file so that I can add other containers to the same docker network and not expose the ports on the host, especially useful on servers with public IPs, however this might be a commented section in the file by default.

**Comment by hobbes444** [comment]
@eyduh I am not entirely following the volume declaration.
Shouldn't it be something like:
```
volumes:
    - ./protonmail:/root
```

**Comment by eyduh** [comment]
For the containers, or services as they technically are referred to in a compose file, there can be a section declaring where to store user data. This can either be a named volume, like

```
volumes:
    - protonmail:/root
```

In which case the volume will need to be declared in a `volumes:` section in the compose file.

Another way to store user data is using bind mounts like you have declared in your example. If using bind mounts then you do not need to declare a volume separately.

There are also predefined volumes that get declared in the Dockerfile if one knows that a certain directory will contain data that needs to be persisted for the user.

You can read more about the different ways to persist user data [here.](http://stackoverflow.com/questions/47150829/ddg#47152658). It is technically not needed but if one wants to move this container from one host to another using volumes is one way to do it, also to make sure the setup is persisted between updates or rebuilds of the image and container.

Seeing as there is no need to browse the files of this container, and the user files are a bit sensitive to permissions, I used a docker volume rather than a bind mount for my setup. The service is run as root so the gpg key files are generated in /root in this case so persisting /root in a docker volume made sense to me, but there is no one right way to do things c:

**Comment by hobbes444** [comment]
Thank for the very detailed answer @eyduh!

How would one initialize the container with compose? If I initiate using `docker run --rm -it -v protonmail:/root shenxn/protonmail-bridge init` and then try to create a docker compose for it, it does not work, docker compose will try to create a new contain with the same name and complain.

**Comment by Nexion** [comment]
@hobbes444 by default `docker compose` creates new volumes and networks prefixed with the assigned project name. If you'd rather prefer it to use an existing volume, it should be marked with `external: true`:
```yaml
volumes:
  protonmail:
    external: true
```

**Comment by hobbes444** [comment]
Thanks @Nexion @eyduh . I did this:

```shell
docker run --rm -it -v protonmail:/root shenxn/protonmail-bridge init
```

then
```yaml
version: '2.1'

services:
  protonmail-bridge:
    image: shenxn/protonmail-bridge
    container_name: pm_bridge
    ports:
      - 127.0.0.1:1025:25/tcp
      - 127.0.0.1:1143:143/tcp
    restart: unless-stopped
    stdin_open: true 
    tty: true
volumes:
  protonmail:
    external: true
```

but that does not seem to work, I get this error:

`WARN[0000] Failed to add test credentials to keychain    error="pass not initialized: exit status 1: Error: password store is empty. Try \"pass init\".\n" helper="*pass.Pass"`

It seems it's not able to find the data created using the initialization command.

**Comment by Nexion** [comment]
@hobbes444 

```yaml
version: '2.1'

services:
  protonmail-bridge:
    image: shenxn/protonmail-bridge
    container_name: pm_bridge
    ports:
      - 127.0.0.1:1025:25/tcp
      - 127.0.0.1:1143:143/tcp
    volumes: # <---
      - protonmail:/root
    restart: unless-stopped
    stdin_open: true 
    tty: true
volumes:
  protonmail:
    external: true
```

**Comment by shenxn** [comment]
Thanks for the contribution! Sorry for replying late. I'll merge this first and feel free to submit any further improvements
